pages/en/supported-networks/cosmos.mdx

```diff
@@ -0,0 +1,228 @@
+
+There are currently three types of handlers supported for the Cosmos subgraphs:
+
+- Block handlers: run whenever a new block is appended to the chain.
+- [Event](https://docs.cosmos.network/master/core/events.html) handlers: run when a specific event is emitted.
+- [Transaction](https://docs.cosmos.network/master/core/transactions.html) handlers: run when a transaction occurs.
+
+Based on the [official Cosmos documentation](https://docs.cosmos.network/):
+
+> Events are objects that contain information about the execution of the application. They are mainly used by service providers like block explorers and wallets to track the execution of various messages and index transactions.
+
+> Transactions are objects created by end-users to trigger state changes in the application.
+
+## Building a Cosmos subgraph
+
+### Subgraph Dependencies
+
+[graph-cli](https://github.com/graphprotocol/graph-cli) is a CLI tool to build and deploy subgraphs, version `>=0.30.0` is required in order to work with Cosmos subgraphs.
+
+[graph-ts](https://github.com/graphprotocol/graph-ts) is a library of subgraph-specific types, version `>=0.27.0` is required in order to work with Cosmos subgraphs.
+
+### Subgraph Main Components
+
+There are three main key parts or files when it comes to defining a subgraph:
+
+**subgraph.yaml**: a YAML file containing the subgraph manifest, which identifies which events to track and how to process them.
+
+**schema.graphql**: a GraphQL schema that defines what data is stored for your subgraph, and how to query it via GraphQL.
+
+**AssemblyScript Mappings**: [AssemblyScript](https://github.com/AssemblyScript/assemblyscript) code that translates from blockchain data to the entities defined in your schema.
+
+### Subgraph Manifest Definition
+
+The subgraph manifest (`subgraph.yaml`) identifies the data sources for the subgraph, the triggers of interest, and the functions (`handlers`) that should be run in response to those triggers. See below for an example subgraph manifest for a Cosmos subgraph:
+
+```yaml
+specVersion: 0.0.5
+description: Cosmos Subgraph Example
+schema:
+  file: ./schema.graphql # link to the schema file
+dataSources:
+  - kind: cosmos
+    name: CosmosHub
+    network: cosmoshub-4 # This will change for each cosmos-based blockchain. In this case, the example uses the CosmosHub mainnet.
+    source:
+      startBlock: 0 # Required for Cosmos, set this to 0 to start indexing from chain genesis
+    mapping:
+      apiVersion: 0.0.7
+      language: wasm/assemblyscript
+      blockHandlers:
+        - handler: handleNewBlock # the function name in the mapping file
+      eventHandlers:
+        - event: rewards # the type of the event that will be handled
+          handler: handleReward # the function name in the mapping file
+      transactionHandlers:
```

Comment:
Seems that there is no filtering of transactionHandlers for now?

Reply:
Message filtering will come with v1.1

Reply:
👍